_transcripts/050-rust-1.56-1.57.md

```diff
@@ -0,0 +1,1302 @@
+scope, the `Iterator` trait. In the prelude, I mean.
+
+And so, you might ask stuff like, why is it necessary? Well, I guess, you know,
+we could say it's kind of just for neatness because `IntoIterator` is already in
+the prelude, although a bit more concrete use case is, if you're mapping over
+something, and you want to say, you know, I want to map over this thing, that's
+an iterator, you can just map over, say, like the first— using first class
+functions you can say, you know, `Vec::from_iter()` as opposed to having to say,
+you know, make a closure `x` and then call `x.collect()` on that closure. So
+that's a bit nicer.
+
+__Jon__: Yeah, that's true. The next one is something that I know you've been a
+little bit involved with, and I think we should combine the next two, which is
+that `panic!`, macros now always expect format strings, just like `println!`.
+And that there is now reserved syntax for custom identifiers before the pound
+symbol. So think of something like raw strings like `r#` and then double quoted
+strings. Identifiers followed— or immediately preceding a string; think of
+something like byte strings. So if you type `b` and then double quotes what you
+get back is a `u8` slice. And same thing for identifier preceding a character
+literal. So this would be something like `b'x'`, which gives you a `u8` instead
+of a character. Do you want to talk about these two and then how they interact?
+
+__Ben__: Yeah, I'll talk about this. So let's back up real quick. There is
+something coming up in an upcoming version of Rust, which you can preview right
+now. It is called implicit formatting captures. So the idea is, currently, if
+you're writing a `println!` macro and you want to, just like, you have a string
+`"hello"` and a string `foo` and you want to put them together, you might type
+`println!` and then you have to give your thing in quotes, your format
+specifier— your format string, really. And so it would be `"{},{}"`, and then
+you after this— it's hard to code in voice, actually.
+
+__Jon__: Yeah, you're right. You're right.
+
+__Ben__: The point is that in your `println!`, after you had your string, of how
+you want to print it, you would need to say your two variables, like maybe you
+have the string `"hello"`, and a variable `foo` and the string `"world"` and
+variable `bar` you would have to end the `println!` invocation with `, foo, bar`
+and that tells the `println!` macro what to put in your brackets individually.
+And this is kind of a shorthand in fact. And so I think people are probably
+familiar with the brackets in the formatting strings syntax, where you can put,
+like, `:`, and you can put `{:?}` or you can put `{:#?}` or if you're formatting
+integers or floats, there's all kinds of various ways you can say that I want
+to, you know, add some white space here, I want you to pad this thing, I want
+you to round this thing to this many decimal points, et cetera.
```

Comment:
Another place where the formatting got a bit tricky. Please review? (audio 21:14)

Reply:
lgtm :+1: